### PR TITLE
Pushing Bug/categories to development

### PR DESCRIPTION
**Pull request by zvidmarb**
Changes/additions made to implement the categories section:
- Uncommented previous categories code
- Adjusted formatting of category buttons along the screen with a flex container
- Implemented method to toggle buttons and add them to the CERF form
- Fixed cancel button not showing up on tags when hovering over them, implemented functionality for deleting categories
- Implemented ability to add categories on button press (with additional keyboard functionality in the HTML)

To review:
- Added autofocus function on the "Add Category" field so cursor always lands on the new field when the button is pressed (might be out of scope, there could be a safer way of implementing it?)
- Accidentally removed json files that were modified on my end (I accidentally committed the changes and meant to revert them)

### CODE REVIEW

**Review by terrencejames — CHANGES_REQUESTED** [review]
I think you meant to exclude the package json files from the commit, not explicitly delete them 😛

**Comment by terrencejames** [comment]
> I think you meant to exclude the package json files from the commit, not explicitly delete them 😛

Oh I see that you noticed this already in your PR. Are you having trouble putting the files back then?

**Comment by zvidmarb** [comment]
Oh yeah, I have the files in my local repository untracked, but would I want to have them in the commit with the unneeded modifications before completing the pull request?

**Comment by terrencejames** [comment]
> Oh yeah, I have the files in my local repository untracked, but would I want to have them in the commit with the unneeded modifications before completing the pull request?

you wouldnt want them in your commit unless you added a new package (like from npm) that you needed in your branch

i think the distinction here is that you deleted package.json from your repository and then committed that, which will then delete the package.json in development 😨 

what i think you meant to do, is leave package.json untracked/unstaged OR delete/remove it from the commit itself and not your repository, which will preserve the package.json in development

if you look at the "Files changed" tab i think you'll see what i mean, it's explicitly deleting the package.json file (also a 17k line code difference haha)

**Comment by terrencejames** [comment]
> Oh yeah, I have the files in my local repository untracked, but would I want to have them in the commit with the unneeded modifications before completing the pull request?

you wouldnt want them in your commit unless you added a new package (like from npm) that you needed in your branch

i think the distinction here is that you deleted package.json from your repository and then committed that, which will then delete the package.json in development 😨

what i think you meant to do, is leave package.json untracked/unstaged OR delete/remove it from the commit itself and not your repository, which will preserve the package.json in development

if you look at the "Files changed" tab i think you'll see what i mean, it's explicitly deleting the package.json file (also a 17k line code difference haha)

**Comment by zvidmarb** [comment]
> > Oh yeah, I have the files in my local repository untracked, but would I want to have them in the commit with the unneeded modifications before completing the pull request?
> 
> you wouldnt want them in your commit unless you added a new package (like from npm) that you needed in your branch
> 
> i think the distinction here is that you deleted package.json from your repository and then committed that, which will then delete the package.json in development 😨
> 
> what i think you meant to do, is leave package.json untracked/unstaged OR delete/remove it from the commit itself and not your repository, which will preserve the package.json in development
> 
> if you look at the "Files changed" tab i think you'll see what i mean, it's explicitly deleting the package.json file (also a 17k line code difference haha)

I see what you mean. I believe the problem was the command Alvin gave me (git rm --cached) because I thought the files were newly committed files unique to my setup. In reality, the json files were just updated (but probably still specific to my local version?)

How should I go about fixing it so the files are untracked, not deleted? Should I checkout to the commit before deletion?

**Comment by terrencejames** [comment]
> > > Oh yeah, I have the files in my local repository untracked, but would I want to have them in the commit with the unneeded modifications before completing the pull request?
> > 
> > 
> > you wouldnt want them in your commit unless you added a new package (like from npm) that you needed in your branch
> > i think the distinction here is that you deleted package.json from your repository and then committed that, which will then delete the package.json in development 😨
> > what i think you meant to do, is leave package.json untracked/unstaged OR delete/remove it from the commit itself and not your repository, which will preserve the package.json in development
> > if you look at the "Files changed" tab i think you'll see what i mean, it's explicitly deleting the package.json file (also a 17k line code difference haha)
> 
> I see what you mean. I believe the problem was the command Alvin gave me (git rm --cached) because I thought the files were newly committed files unique to my setup. In reality, the json files were just updated (but probably still specific to my local version?)
> 
> How should I go about fixing it so the files are untracked, not deleted? Should I checkout to the commit before deletion?

haha it's okay i literally just did the same thing rn at work so i had to look it up as well😅 

what you're trying to do is revert a specific file to a previous commit: https://stackoverflow.com/questions/215718/how-can-i-reset-or-revert-a-file-to-a-specific-revision

**Comment by terrencejames** [comment]
Are you using the command line for git btw? I highly recommend GitKraken or another GitGUI since it really helps make things visually cleaner and its better than having to do git log or git diff constantly, etc and u can see exactly what files are changed/staged

**Comment by zvidmarb** [comment]
> Are you using the command line for git btw? I highly recommend GitKraken or another GitGUI since it really helps make things visually cleaner and its better than having to do git log or git diff constantly, etc and u can say exactly what files are changed/staged

I'm switching between Github Desktop and the git functionalities in VSCode but I might try out one of those to see if it's any better haha.

As for the .json files, it looks I can actually add the files back really easily on Github desktop. When you mean untracking my changes to them, would adding them to .gitignore achieve that (Github desktop seems to do it automatically for me)?

**Comment by terrencejames** [comment]
> > Are you using the command line for git btw? I highly recommend GitKraken or another GitGUI since it really helps make things visually cleaner and its better than having to do git log or git diff constantly, etc and u can say exactly what files are changed/staged
> 
> I'm switching between Github Desktop and the git functionalities in VSCode but I might try out one of those to see if it's any better haha.
> 
> As for the .json files, it looks I can actually add the files back really easily on Github desktop. When you mean untracking my changes to them, would adding them to .gitignore achieve that (Github desktop seems to do it automatically for me)?

oh, VS has a good git UI too, i use gitkraken cause i use atom as my text editor

also sorry, maybe untrack might not have been the best word because it's a specific technical term. you just wanted it unstaged. so putting it in the .gitignore might not be the best method because you would be committing the gitignore and saying to ignore all future changes to this file (which would be a problem in case we wanted to add new packages to the package.json)

File: src/app/modules/cerf/cerf.component.ts [hunk 1]
```diff
@@ -239,6 +239,37 @@ export class CerfComponent {
...
+		//label already exists check
+		if (!this.categoryButtons.includes(label)) {
+			this.categoryButtons.push(label);
+		}
+		this.addingCategory = false;
+	}
+
+	//deletes existing labels from the array of categories
+	removeLabel(index: number): void {
+		this.categoryButtons.splice(index);
```
Comment (terrencejames) [inline src/app/modules/cerf/cerf.component.ts]: I think the rest of this looks good. But I just noticed you're using splicing to remove elements of an array. Won't this cut off everything after the index as well? So like, if you had [0,1,2,3,4,5] and splice off index 2, it'd just be [0,1]?

**Review by terrencejames — CHANGES_REQUESTED** [review]
Left some comments about array splicing, feel free to ignore if i'm wrong

File: src/app/modules/cerf/cerf.component.ts (hunk 1 above)
Comment (terrencejames) [reply src/app/modules/cerf/cerf.component.ts]: in the function above this i think you use splicing as well, should make sure that one behaves alright too

**Review by terrencejames — DISMISSED** [review]
nice

**Review by terrencejames — CHANGES_REQUESTED** [review]
i noticed some code was committed thats only relevant to a different bug, try to see if you can revert it to just right after i first approved the PR

**Review by terrencejames — APPROVED** [review]

**Pull request merged**